dm/pkg/binlog/event/common.go

```diff
@@ -207,7 +211,9 @@ func verifySingleGTID(flavor string, gSet gmysql.GTIDSet) (interface{}, error) {
 			return nil, terror.ErrBinlogOnlyOneGTIDSupport.Generate(len(mariaGTIDs.Sets), gSet)
 		}
 		var mariaGTID *gmysql.MariadbGTID
-		for _, mariaGTID = range mariaGTIDs.Sets {
+		for _, set := range mariaGTIDs.Sets {
+			for _, mariaGTID = range set {
```

Comment:
check `len` is 1

Reply:
we can check len(set) before the for loop

Reply:
That can't get the right gtid numbers in the error message